__tests__/app.test.js

```diff
@@ -37,13 +38,56 @@ describe('GET /api/topics', () => {
 			.get('/api/topics')
 			.expect(200)
 			.then(({ body }) => {
-				for (item of body) {
-					expect(item).toMatchObject({
+				const { topics } = body;
+				expect(topics.length).toBe(3);
+				for (const topic of topics) {
+					expect(topic).toMatchObject({
 						slug: expect.any(String),
 						description: expect.any(String),
 					});
 				}
-				expect(body.length).toBe(3);
+			});
+	});
+});
+
+describe('GET /api/articles/:article_id', () => {
+	test('200: return an article when a valid id is supplied', () => {
+		const article_id = 3;
+		return request(app)
+			.get(`/api/articles/${article_id}`)
+			.expect(200)
+			.then(({ body }) => {
+				const { articles } = body;
+				for (const article of articles) {
+					expect(article).toMatchObject({
+						article_id: expect.any(Number),
+						title: expect.any(String),
+						topic: expect.any(String),
+						author: expect.any(String),
+						body: expect.any(String),
+						created_at: expect.any(String),
+						votes: expect.any(Number),
+						article_img_url: expect.any(String),
+					});
+				}
+			});
+	});
+	test('404: responds with an error if the article does not exist', () => {
+		const article_id = 30;
+		return request(app)
+			.get(`/api/articles/${article_id}`)
+			.expect(404)
+			.then(({ body }) => {
+				expect(body.msg).toBe('Not found.');
+			});
+	});
+	test('404: responds with an error if the article id is not vaild', () => {
```

Comment:
Be careful that you test descriptions match what is being asserted (when cutting and pasting! 😄 )